sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/MicrosoftTeamsUserIdentifier.java

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.common;
+
+import com.azure.android.core.util.CoreUtil;
+
+/**
+ * Communication identifier for Microsoft Teams User
+ */
+public class MicrosoftTeamsUserIdentifier extends CommunicationIdentifier {
+
+    private final String userId;
+    private final boolean isAnonymous;
+    private String id;
+    private CommunicationCloudEnvironment cloudEnvironment = CommunicationCloudEnvironment.PUBLIC;
+
+    /**
+     * Creates a MicrosoftTeamsUserIdentifier object
+     *
+     * @param userId the string identifier representing the identity
+     * @param isAnonymous set this to true if the user is anonymous,
+     *                    for example when joining a meeting with a share link
+     * @throws IllegalArgumentException thrown if id parameter fail the validation.
+     */
+    public MicrosoftTeamsUserIdentifier(String userId, boolean isAnonymous) {
```

Comment:
Let's make sure that we explain exactly what `id` and `userId` are and how they're different. You can copy over the comments verbatim from C#.

https://github.com/Azure/azure-sdk-for-net/blob/84dd28fc7ff4315e7cbe81e27cbdcbb1c1cbe849/sdk/communication/Azure.Communication.Common/src/MicrosoftTeamsUserIdentifier.cs#L26-L29